cpp/tests/dictionary/set_keys_test.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/dictionary/dictionary_column_view.hpp>
+#include <cudf/dictionary/encode.hpp>
+#include <cudf/dictionary/update_keys.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+#include <vector>
+
+struct DictionarySetKeysTest : public cudf::test::BaseFixture {};
+
+TEST_F(DictionarySetKeysTest, StringsKeys)
+{
+    cudf::test::strings_column_wrapper strings{ "eee", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "eee", "aaa" };
+    auto dictionary = cudf::dictionary::encode( strings );
+
+    cudf::test::strings_column_wrapper new_keys{ "aaa", "ccc", "eee", "fff" };
+    auto result = cudf::dictionary::set_keys( dictionary->view(), new_keys );
+
+    cudf::dictionary_column_view view(result->view());
+    cudf::test::expect_columns_equal(view.keys(), new_keys);
+    cudf::test::fixed_width_column_wrapper<int32_t> indices_expected{2,0,2,1,1,1,1,2,0};
```

Comment:
Where do the 3rd and 4th indices here come from? These two {2,0,`2,1`,1,1...} I think correspond to "ddd" and "bbb" above, which aren't in `new_keys` so I'd expect these to be null?

Reply:
Yes, you are correct. I need to fix the tests to check for nulls properly and not undefined values.